src/libvirtApi/domain.js

```diff
@@ -567,6 +567,23 @@ export function domainDetachIface({ connectionName, index, vmName, live, persist
     return cockpit.spawn(args, options);
 }
 
+export function domainRemoveVsock({ connectionName, vmName, permanent, hotplug }) {
+    const args = ['virt-xml', '-c', `qemu:///${connectionName}`, vmName, '--remove-device', '--vsock', 1];
```

Comment:
What is that `1`? For spawn this must all be strings, no?

Reply:
VM can have only 1 vsock attached to it, so this says to remove the one and only vsock on the machine.

Reply:
I can change it to string as `"1"` so it's not confusing, if you want

Reply:
Fixed

Reply:
Yes, that's what I meant -- maybe there's some magic somewhere to convert all spawn arguments to strings, but it's confusing and brittle. Thanks!

Reply:
Note you didn't change it to `"1"`, but  dropped the argument entirely. That may explain the "ERROR    XML error: only a single vsock device is supported" test failure?